examples/spmm/spmm.cc

```diff
@@ -292,11 +292,16 @@ class SpMM25D {
       , parallel_bcasts_(parallel_bcasts) {
     Edge<Key<2>, void> a_ctl, b_ctl;
     Edge<Key<2>, int> a_rowctl, b_colctl; // TODO: can we have multiple control inputs per TT?
+    auto constraint = ttg::make_shared_constraint<ttg::SequencedKeysConstraint<Key<2>>>();
```

Comment:
I agree. I have not found a good way to type pun the keys. Today, we couldn't include the gemm task in the constraint because the key types are different (`Key<2>` and `Key<3>`). I'm sure there is way to do that, I'm not sure about the complexity...